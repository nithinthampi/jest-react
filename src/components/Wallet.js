import React, { Component } from "react";
import { connect } from "react-redux";
import { deposit, withdraw } from "../actions/balance";

export class Wallet extends Component {
  constructor() {
    super();
    this.state = { amount: 0 };
  }

  render() {
    return (
      <div>
        <h3 className="balance">Balance:{this.props.balance}</h3>
        <input
          className="input-amount"
          onChange={event => {
            this.setState({ amount: parseInt(event.target.value) });
          }}
        />
        <button
          className="btn-deposit"
          onClick={() => this.props.deposit(this.state.amount)}
        >
          deposit
        </button>
        <button
          className="btn-withdraw"
          onClick={() => this.props.withdraw(this.state.amount)}
        >
          withdraw
        </button>
      </div>
    );
  }
}

export default connect(
  state => {
    return { balance: state };
  },
  { deposit, withdraw }
)(Wallet);
