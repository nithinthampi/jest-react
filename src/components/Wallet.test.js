import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Wallet } from "./Wallet";

configure({ adapter: new Adapter() });

describe("Wallet", () => {
  const balance = 0;
  const deposit = jest.fn();
  const withdraw = jest.fn();
  const props = { balance, deposit, withdraw };
  const wallet = shallow(<Wallet {...props} />);
  it("renders propely", () => {
    expect(wallet).toMatchSnapshot();
  });

  it("Renders the intial balance correctly", () => {
    expect(wallet.find(".balance").text()).toEqual(`Balance:${balance}`);
  });

  describe("when user enters amount", () => {
    beforeEach(() => {
      wallet
        .find(".input-amount")
        .simulate("change", { target: { value: 10 } });
    });

    it("sets the amount in wallte component `state`", () => {
      expect(wallet.state().amount).toEqual(10);
    });

    describe("And user clicks on `deposit` button", () => {
      beforeEach(() => {
        wallet.find(".btn-deposit").simulate("click");
      });
      it("Updates the deposit in balance", () => {
        expect(deposit).toHaveBeenCalledWith(10);
      });
    });

    describe("And the user clicks on `withdraw` button", () => {
      beforeEach(() => {
        wallet.find(".btn-withdraw").simulate("click");
      });

      it("Calls the withdraw methos with right parameters", () => {
        expect(withdraw).toHaveBeenCalledWith(10);
      });
    });
  });
});
