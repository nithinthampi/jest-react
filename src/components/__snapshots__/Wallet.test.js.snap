// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders propely 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={0}
    deposit={[Function]}
    withdraw={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="balance"
>
          Balance:
          0
</h3>,
        <input
          className="input-amount"
          onChange={[Function]}
/>,
        <button
          className="btn-deposit"
          onClick={[Function]}
>
          deposit
</button>,
        <button
          className="btn-withdraw"
          onClick={[Function]}
>
          withdraw
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Balance:",
            0,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Balance:",
          0,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-amount",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "deposit",
          "className": "btn-deposit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "deposit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "withdraw",
          "className": "btn-withdraw",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "withdraw",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="balance"
>
            Balance:
            0
</h3>,
          <input
            className="input-amount"
            onChange={[Function]}
/>,
          <button
            className="btn-deposit"
            onClick={[Function]}
>
            deposit
</button>,
          <button
            className="btn-withdraw"
            onClick={[Function]}
>
            withdraw
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Balance:",
              0,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Balance:",
            0,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-amount",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "deposit",
            "className": "btn-deposit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "deposit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "withdraw",
            "className": "btn-withdraw",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "withdraw",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
